srcs/client/transmit_message.c

```diff
@@ -43,18 +49,18 @@ static void	transmit_bit(pid_t pid, char bit)
 
 static void	transmit_byte(pid_t pid, char c)
 {
-	int							bit_index;
-	size_t						attempt_count;
-	t_signal_acknowledgement	ack_status;
+	int				bit_index;
+	size_t			attempt_count;
+	sig_atomic_t	ack_status;
 
 	bit_index = 7;
 	attempt_count = 0;
 	while (bit_index >= 0)
 	{
-		g_client_info.signal_status = ACK_WAITING;
+		g_signal_pid_state = pid;
 		transmit_bit(pid, (c >> bit_index) & 1);
 		attempt_count++;
-		ack_status = is_ack_received();
+		ack_status = get_ack_status();
 		if (ack_status == ACK_RECEIVED)
 		{
 			bit_index--;
```

Comment:
Should I avoid using `sig_atomic_t` here because `sig_atomic_t ack_status` is merely a variable to hold a copy of a flag? The variable that the signal handler accesses is the flag itself, and a copy of it will never be accessed by the signal handler. What do you think?